oak_utils/src/lib.rs

```diff
@@ -151,12 +163,10 @@ impl prost_build::ServiceGenerator for OakServiceGenerator {
 
             #[allow(clippy::unit_arg)]
             impl <T: #service_name> #oak_package::grpc::ServerNode for #dispatcher_name<T> {
-                fn invoke(&mut self, method: &str, req: &[u8], writer: #oak_package::grpc::ChannelResponseWriter) {
-                    match method {
-                        #(#dispatcher_methods,)*
-                        _ => {
-                            ::log::error!("unknown method name: {}", method);
-                        }
+                fn invoke(&mut self, method_name: &str, req: &[u8], writer: #oak_package::grpc::ChannelResponseWriter) {
+                    match #dispatcher_name::<T>::server_method(method_name) {
+                        Some(server_method) => server_method.invoke(&mut self.0, req, writer),
+                        None => ::log::error!("unknown method name: {}", method_name),
```

Comment:
This ends up looking a bit odd/superfluous in the generated code &ndash; the generated file for every proto is going to have exactly the same `impl` for `invoke()`, which probably means this PR (or a follow-up) could go a step or two further.

Maybe something like:
 - Add a new `oak::grpc::ServerMethodLookup` trait, with a single entry `fn server_method(method_name: &str) -> Option<&dyn ::oak::grpc::Invocable<T>>`.
 - Change the code generation so that `server_method` is inside a `impl <T: #service_name> oak::grpc::ServerMethodLookup for  #dispatcher_name<T> { ... }` block.
 - Add a default `impl` of `oak::grpc::ServerNode` for anything with trait bound `oak::grpc::ServerMethodLookup` that has this common implementation of `invoke()`.

(I also wonder whether the generated `impl`s of `Default` and `oak::WithInit` can be moved out of the generated code &ndash; it would be nice to just have generated code that's specific to the particular service, rather than having copies of the same code in every generated file.)

Reply:
Yes I am planning follow up PRs to basically remove the `ServerNode` trait entirely, and ideally also clean up the `Default` and `WithInit`, though the orphan and coherence rules may make that a bit difficult to clean up entirely.